aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -58,6 +71,78 @@ public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration} by reading dataStore
+     * settings from the config file.
+     * @param context application context which will be used to retrieve amplicationconfiguration.json
+     * @param pluginConfigKey the desired dataStore plugin config key (i.e. awsDataStorePlugin)
+     * @return A new builder instance
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    public static Builder builder(@NonNull Context context, @NonNull String pluginConfigKey) throws DataStoreException {
```

Comment:
is `Context` require at this point? I think of the configuration object should be a simple pojo. It won't execute any configuration logic, but just carry their values. So the context should be used when the Plugin's `configure(JSONObject pluginConfiguration, @NonNull Context context)` is invoked. Your pojo is available to execute any configuration workflow you need at that moment of the plugin lifecycle.